File: backend/src/models/PostMedia.js
const { DataTypes } = require('sequelize');
const sequelize = require('../config/database');
const Post = require('./Post');

const PostMedia = sequelize.define('PostMedia', {
  id: {
    type: DataTypes.UUID,
    defaultValue: DataTypes.UUIDV4,
    primaryKey: true
  },
  post_id: {
    type: DataTypes.UUID,
    allowNull: false,
    references: {
      model: 'posts',
      key: 'id'
    },
    onDelete: 'CASCADE'
  },
  media_type: {
    type: DataTypes.STRING(20),
    allowNull: false,
    // 'image', 'video', etc.
  },
  media_url: {
    type: DataTypes.TEXT,
    allowNull: false
  },
  thumbnail_url: {
    type: DataTypes.TEXT,
    allowNull: true
  },
  created_at: {
    type: DataTypes.DATE,
    defaultValue: DataTypes.NOW
  }
}, {
  tableName: 'post_media',
  timestamps: false
});

// Define association
PostMedia.belongsTo(Post, { foreignKey: 'post_id' });
Post.hasMany(PostMedia, { foreignKey: 'post_id' });

module.exports = PostMedia;

File: backend/app/models/post_media.py
from datetime import datetime
from app import db

class PostMedia(db.Model):
    __tablename__ = 'post_media'
    
    id = db.Column(db.Integer, primary_key=True)
    post_id = db.Column(db.Integer, db.ForeignKey('posts.id'), nullable=False)
    media_type = db.Column(db.String(20), nullable=False)  # image, video, etc.
    media_url = db.Column(db.String(255), nullable=False)
    created_at = db.Column(db.DateTime, default=datetime.utcnow)
    updated_at = db.Column(db.DateTime, default=datetime.utcnow, onupdate=datetime.utcnow)
    
    def to_dict(self):
        return {
            'id': self.id,
            'post_id': self.post_id,
            'media_type': self.media_type,
            'media_url': self.media_url,
            'created_at': self.created_at.isoformat(),
            'updated_at': self.updated_at.isoformat()
        }
